src/Query/index.ts

```diff
@@ -179,7 +180,7 @@ export class Query {
       },
     });
 
-    const compound: { should: any[]; must: any[]; filter?: any[]; minimumShouldMatch: number } = {
+    const compound: { should: CompoundPart[]; must: Must[]; filter?: any[]; minimumShouldMatch: number } = {
```

Comment:
i left the `filter` parameter as `any[]` because I could not see where it is used.